doc/dev/kclient_dev_bootstrap.rst

```diff
@@ -0,0 +1,474 @@
+development can lead to system crashes, data loss, or even hardware damage. By
+using VMs, we isolate the kernel environment from the dev machine's operating
+system, allowing testing of potentially unstable code without taking undue
+risks. Using VMs accelerates development iterations by enabling rapid testing
+and easy rollback through snapshots, allowing developers to quickly test
+changes and revert to previous states without lengthy setup times.
+Additionally, VMs provide controlled hardware emulation, enabling the testing
+of kernel functionality across various simulated architectures and
+configurations (e.g., multiple NICs or storage devices).
+
+This document focuses on using `QEMU <https://www.qemu.org/docs/master/>`_,
+which is available on Linux and WSL2 (Windows). The principles discussed here
+apply to native Windows and to macOS environments as well.
+
+Installing a QEMU VM Using an ISO Image
+---------------------------------------
+
+Preparation
+^^^^^^^^^^^
+
+To install a virtual machine in QEMU using an ISO image, first ensure that QEMU
+is installed on your system. Tool installation is beyond the scope of this
+document and will vary depending on your Linux distribution, so consult your
+package manager’s documentation to install the necessary QEMU packages.
+
+Download the ISO and Create a Virtual Disk
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Begin by downloading the ISO file of the operating system you wish to install,
+such as a Linux distribution like Ubuntu, Fedora, or CentOS. After the ISO is
+ready, create a virtual disk for the OS installation using `qemu-img`, which
+allows for flexible disk management and supports features like snapshots. You
+can create a 128 GB virtual disk in the QCOW2 format with the following
+command:
+
+.. prompt:: bash #
+
+   qemu-img create -f qcow2 my_vm_disk.qcow2 128G
+
+Boot and Install the OS
+Next, boot the VM from the ISO to install the OS by using the `qemu-system` command, specifying both the ISO and the virtual disk:
+
+.. code-block:: bash
+
+        qemu-system-x86_64    -m 2048                                   \
+                        -enable-kvm                                     \
+                        -boot d                                         \
+                        -cdrom /path/to/your.iso                        \
+                        -drive file=my_vm_disk.qcow2,format=qcow2       \
+                        -vnc :1
+
+In this command:
+
+- ``-m 2048`` allocates 2 GB of memory.
+- ``-enable-kvm`` enables hardware acceleration for better performance if
+  available.
+- ``-boot d`` specifies booting from the CD-ROM drive (the ISO image).
+- ``-cdrom /path/to/your.iso`` provides the path to the ISO.
+- ``-drive file=my_vm_disk.qcow2,format=qcow2`` links the virtual disk file.
+- ``-vnc :1`` enables VNC access on display `1`, which corresponds to
+  ``localhost:5901`` (for VNC viewers, this address would be ``localhost:1``).
```

Comment:
Let's move this step to a separate doc. Move relevants parts of step 2:

https://docs.ceph.com/en/latest/dev/kclient/#step-two-create-a-vm

to that same doc. Indicate there are two approaches. (I think yours is more flexible but requires effort to actually install the OS using its procedures on the `my_vm_disk.qcow2` file.)